File: app/buildz/tmp/kapt3/stubs/debug/com/bits/bee/bpmc/presentation/ui/sign_up/info_bisnis/InfoBisnisViewModel.java
package com.bits.bee.bpmc.presentation.ui.sign_up.info_bisnis;

import java.lang.System;

@dagger.hilt.android.lifecycle.HiltViewModel()
@kotlin.Metadata(mv = {1, 7, 1}, k = 1, d1 = {"\u0000F\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0010 \n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0004\b\u0007\u0018\u00002\u000e\u0012\u0004\u0012\u00020\u0002\u0012\u0004\u0012\u00020\u00030\u0001:\u0001\u0016B\u0017\b\u0007\u0012\u0006\u0010\u0004\u001a\u00020\u0005\u0012\u0006\u0010\u0006\u001a\u00020\u0007\u00a2\u0006\u0002\u0010\bJ\u0006\u0010\u0010\u001a\u00020\u0011J\u0018\u0010\u0012\u001a\u0014\u0012\u0010\u0012\u000e\u0012\n\u0012\b\u0012\u0004\u0012\u00020\r0\f0\u000b0\u0013J\u0012\u0010\u0014\u001a\u000e\u0012\n\u0012\b\u0012\u0004\u0012\u00020\u000f0\u000b0\u0013J\u0006\u0010\u0015\u001a\u00020\u0011R \u0010\t\u001a\u0014\u0012\u0010\u0012\u000e\u0012\n\u0012\b\u0012\u0004\u0012\u00020\r0\f0\u000b0\nX\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u000e\u0010\u0004\u001a\u00020\u0005X\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u000e\u0010\u0006\u001a\u00020\u0007X\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u001a\u0010\u000e\u001a\u000e\u0012\n\u0012\b\u0012\u0004\u0012\u00020\u000f0\u000b0\nX\u0082\u000e\u00a2\u0006\u0002\n\u0000\u00a8\u0006\u0017"}, d2 = {"Lcom/bits/bee/bpmc/presentation/ui/sign_up/info_bisnis/InfoBisnisViewModel;", "Lcom/bits/bee/bpmc/presentation/base/BaseViewModel;", "Lcom/bits/bee/bpmc/presentation/ui/sign_up/info_bisnis/InfoBisnisState;", "Lcom/bits/bee/bpmc/presentation/ui/sign_up/info_bisnis/InfoBisnisViewModel$UIEvent;", "cityInteractor", "Lcom/bits/bee/bpmc/domain/usecase/CityInteractor;", "postSignUpUseCase", "Lcom/bits/bee/bpmc/domain/usecase/signup/PostSignUpUseCase;", "(Lcom/bits/bee/bpmc/domain/usecase/CityInteractor;Lcom/bits/bee/bpmc/domain/usecase/signup/PostSignUpUseCase;)V", "cityEntityResponse", "Landroidx/lifecycle/MediatorLiveData;", "Lcom/bits/bee/bpmc/utils/Resource;", "", "Lcom/bits/bee/bpmc/domain/model/City;", "registerResponse", "Lcom/bits/bee/bpmc/data/data_source/remote/response/SignUpResponse;", "getCityList", "", "observeListCity", "Landroidx/lifecycle/LiveData;", "observeSignUpResponse", "postSignUp", "UIEvent", "app_debug"})
public final class InfoBisnisViewModel extends com.bits.bee.bpmc.presentation.base.BaseViewModel<com.bits.bee.bpmc.presentation.ui.sign_up.info_bisnis.InfoBisnisState, com.bits.bee.bpmc.presentation.ui.sign_up.info_bisnis.InfoBisnisViewModel.UIEvent> {
    private final com.bits.bee.bpmc.domain.usecase.CityInteractor cityInteractor = null;
    private final com.bits.bee.bpmc.domain.usecase.signup.PostSignUpUseCase postSignUpUseCase = null;
    private androidx.lifecycle.MediatorLiveData<com.bits.bee.bpmc.utils.Resource<java.util.List<com.bits.bee.bpmc.domain.model.City>>> cityEntityResponse;
    private androidx.lifecycle.MediatorLiveData<com.bits.bee.bpmc.utils.Resource<com.bits.bee.bpmc.data.data_source.remote.response.SignUpResponse>> registerResponse;
    
    @javax.inject.Inject()
    public InfoBisnisViewModel(@org.jetbrains.annotations.NotNull()
    com.bits.bee.bpmc.domain.usecase.CityInteractor cityInteractor, @org.jetbrains.annotations.NotNull()
    com.bits.bee.bpmc.domain.usecase.signup.PostSignUpUseCase postSignUpUseCase) {
        super();
    }
    
    @org.jetbrains.annotations.NotNull()
    public final androidx.lifecycle.LiveData<com.bits.bee.bpmc.utils.Resource<java.util.List<com.bits.bee.bpmc.domain.model.City>>> observeListCity() {
        return null;
    }
    
    public final void getCityList() {
    }
    
    @org.jetbrains.annotations.NotNull()
    public final androidx.lifecycle.LiveData<com.bits.bee.bpmc.utils.Resource<com.bits.bee.bpmc.data.data_source.remote.response.SignUpResponse>> observeSignUpResponse() {
        return null;
    }
    
    public final void postSignUp() {
    }
    
    @kotlin.Metadata(mv = {1, 7, 1}, k = 1, d1 = {"\u0000\f\n\u0002\u0018\u0002\n\u0002\u0010\u0000\n\u0002\b\u0002\b6\u0018\u00002\u00020\u0001B\u0007\b\u0004\u00a2\u0006\u0002\u0010\u0002\u00a8\u0006\u0003"}, d2 = {"Lcom/bits/bee/bpmc/presentation/ui/sign_up/info_bisnis/InfoBisnisViewModel$UIEvent;", "", "()V", "app_debug"})
    public static abstract class UIEvent {
        
        private UIEvent() {
            super();
        }
    }
}

File: app/src/main/java/com/bits/bee/bpmc/presentation/ui/sign_up/info_bisnis/InfoBisnisViewModel.kt
package com.bits.bee.bpmc.presentation.ui.sign_up.info_bisnis

import android.content.Context
import androidx.lifecycle.LiveData
import androidx.lifecycle.MediatorLiveData
import com.bits.bee.bpmc.data.data_source.remote.response.BidangUsahaResponse
import com.bits.bee.bpmc.data.data_source.remote.response.SignUpResponse
import com.bits.bee.bpmc.domain.model.Utm
import com.bits.bee.bpmc.domain.usecase.common.GetLatestCityUseCase
import com.bits.bee.bpmc.domain.usecase.signup.GetBidangUsahaUseCase
import com.bits.bee.bpmc.domain.usecase.signup.PostSignUpUseCase
import com.bits.bee.bpmc.presentation.base.BaseViewModel
import com.bits.bee.bpmc.utils.BeePreferenceManager
import com.bits.bee.bpmc.utils.Resource
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.flow
import javax.inject.Inject


@HiltViewModel
class InfoBisnisViewModel @Inject constructor(
    private val getLatestCityUseCase: GetLatestCityUseCase,
    private val postSignUpUseCase: PostSignUpUseCase,
    private val getBidangUsahaUseCase: GetBidangUsahaUseCase
) : BaseViewModel<InfoBisnisState,InfoBisnisViewModel.UIEvent >() {

    init {
        state = InfoBisnisState()
    }

    var usahaList = mutableListOf<BidangUsahaResponse.DataUsaha>()

    var kategoriList = mutableListOf<String>()

    val cityList = getLatestCityUseCase(true)

    val bidangUsahaList = getBidangUsahaUseCase()

    private var registerResponse: MediatorLiveData<Resource<SignUpResponse>> = MediatorLiveData()
    fun observeSignUpResponse() = registerResponse as LiveData<Resource<SignUpResponse>>

    suspend fun postSignUp(context : Context) : Flow<Resource<SignUpResponse>> {
        state.signUp?.let {
            val utm = Utm(
                "ANDROID"
            )

            val utmData: String = BeePreferenceManager.getDataFromPreferences(context, "utm", "") as String
            val utms: Array<String>

            val utmList: MutableList<String> = ArrayList()
            utmList.add("utm_source")
            utmList.add("utm_medium")
            utmList.add("utm_campaign")
            utmList.add("utm_term")
            utmList.add("utm_content")
            if (utmData.isNotEmpty()) {
                utms = utmData.split("&".toRegex()).toTypedArray()
                if (utms.isNotEmpty()) {
                    for (i in utms.indices) {
                        if (utms[i].contains("utm_source")) {
                            val utmVal = utms[i].split("=".toRegex()).toTypedArray()
                            utm. utm_source = if (utmVal[1] == "google-play") "ANDROID" else "ANDR-ADS"
                        }
                        if (utms[i].contains("utm_medium")) {
                            val utmVal = utms[i].split("=".toRegex()).toTypedArray()
                            utm.utm_medium = utmVal[1]
                        }
                        if (utms[i].contains("utm_campaign")) {
                            val utmVal = utms[i].split("=".toRegex()).toTypedArray()
                            utm.utm_campaign = utmVal[1]
                        }
                        if (utms[i].contains("utm_term")) {
                            val utmVal = utms[i].split("=".toRegex()).toTypedArray()
                            utm.utm_term = utmVal[1]
                        }
                        if (utms[i].contains("utm_content")) {
                            val utmVal = utms[i].split("=".toRegex()).toTypedArray()
                            utm.utm_content = utmVal[1]
                        }
                    }
                }
            }

            val signUp = it.copy(
                namaPerusahaan = state.namaPerusahaan,
                address = state.alamat,
                bidangUsah = usahaList.firstOrNull { it.name == state.tipeUsaha }?.id.toString(),
                step = 2,
                utm_content = utm.utm_content,
                utm_source = utm.utm_source,
                utm_campaign = utm.utm_campaign,
                utm_medium = utm.utm_medium,
                utm_term = utm.utm_term,
            )
            return postSignUpUseCase(signUp, state.kota)
        }
        return flow {  }
    }

    sealed class UIEvent {

    }

}
//{"address":"Tes","bussiness":"5","city_id":"1","company":"Tes","cust_id":0,"dataUTM":{"utm_campaign":"","utm_content":"","utm_medium":"","utm_source":"ANDROID","utm_term":""},"email":"hafis@beeaccounting.comm","mobile":"08454554588","otp_tipe":"WA","regid":"46649","step":2,"username":"Tes"}